text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,678 @@
+  targets, subject to reasonable target limitations.
+  - Adding a substantively different interface to an existing tool, or a
+    target-specific interface to the functionality of an existing tool,
+    requires design and implementation approval (e.g. RFC/MCP) from the
+    appropriate approving teams for that tool.
+    - Such an interface should have a design that could potentially work for
+      other targets with similar properties.
+    - This should happen separately from the review and approval of the target,
+      to simplify the target review and approval processes, and to simplify the
+      review and approval processes for the proposed new interface.
+  - By way of example, a target that runs within a sandbox may need to modify
+    the handling of files, tool invocation, and similar to meet the
+    expectations and conventions of the sandbox, but must not introduce a
+    separate "sandboxed compilation" interface separate from the CLI interface
+    without going through the normal approval process for such an interface.
+    Such an interface should take into account potential other targets with
+    similar sandboxes.
+- If the host tools for the platform would normally be expected to be signed or
+  equivalent (e.g. if running unsigned binaries or similar involves a
+  "developer mode" or an additional prompt), it must be possible for the Rust
+  project's automated builds to apply the appropriate signature process,
+  without any manual intervention by either Rust developers, target
+  maintainers, or a third party.  This process must meet the approval of the
+  infrastructure team.
+  - This process may require one-time or semi-regular manual steps by the
+    infrastructure team, such as registration or renewal of a signing key. Any
+    such manual process must meet the approval of the infrastructure team.
+  - This requirement exists to ensure that Rust builds, including nightly
+    builds, can meet the necessary requirements to allow users to smoothly run
+    the host tools.
+- Providing host tools does not exempt a target from requirements to support
+  cross-compilation if at all possible.
+- All requirements for tier 2 apply.
+
+A target may be promoted directly from tier 3 to tier 2 with host tools if it
+meets all the necessary requirements, but doing so may introduce substantial
+additional complexity. If in doubt, the target should qualify for tier 2
+without host tools first.
+
+## Tier 1 target policy
+
+At this tier, the Rust project guarantees that a target builds and passes all
+tests, and will reject patches that fail to build or pass the testsuite on a
+target. We hold tier 1 targets to our highest standard of requirements.
+
+A proposed new tier 1 target must be reviewed and approved by the compiler team
+based on these requirements. In addition, the release team must approve the
+long-term viability of the target, and the additional work of supporting the
```

Comment:
IMO, the additional work of supporting this target is best evaluated by the library and compiler teams, not the release team, which does little to no development work. Perhaps I'm misunderstanding what work is being referenced here.

Reply:
I've clarified this. The intention is for the release team to make the call on the value of supporting the platform (and likewise for the corresponding host tool requirement), not the amount of work required.

As an example, the release team could say "this target just doesn't seem like it's worth having host tools on; yes, it's *possible*, but it seems more like a clever stunt rather than something people will actually use in production". That's orthogonal to the compiler team evaluating maintainability and similar.

This is somewhat related to the requirement about "substantial, widespread interest". This statement grew out of a desire to provide more specificity about which teams are evaluating which aspects of which requirements.